exogenous/model.py

```diff
@@ -1,149 +1,244 @@
 #!/usr/bin/env python3
 
 
-# This Python module models an exogenous-driven land transport scenario involving ICE/BE-based vehicles (graphically represented in
+# This Python module implements an exogenous-driven land transport (toy) model involving ICE/BE-based vehicles (represented in
 # https://github.com/ricnogfer/pypsa-eur-sec_land_transport/blob/master/resources/docs/exogenous_endogenous_land_transport.pdf)
 
 
 
+# TODO:
+# 1) fix solar profile
+# 2) implement CO2 constraint
+# 3) enable BE-vehicles to behave like stores (i.e. to send back electricity to the bus "electricity")
+
+
+
 # import necessary packages
+import os
 import sys
-import yaml
 import pypsa
+import pandas
 
 
 
-# function to create PyPSA-based network from scratch
-def create_network(oil_capital_cost, oil_marginal_cost, solar_capital_cost, solar_marginal_cost, ice_share, bev_share):
+# function to create PyPSA network
+def create_network(parameters, snapshots):
     """
     Parameters
     ----------
-    oil_capital_cost : float
-        Capital cost of oil generator
-    oil_marginal_cost : float
-        Marginal cost of oil generator
-    solar_capital_cost : float
-        Capital cost of solar generator
-    solar_marginal_cost : float
-        Marginal cost of solar generator
-    ice_share : Panda dataframe
-        ICE-vehicle share values for horizon planning
-    bev_share : Panda dataframe
-        BE-vehicle share values forhorizon planning
+    parameters : a Python dictionary (dict())
+        Parameters needed to create the PyPSA network.
+    snapshots : a Pandas date/time series (pandas.core.indexes.datetimes.DatetimeIndex())
+        Number of snapshots that the PyPSA network has/will solve.
 
     Returns
     -------
-    A Network object representing a PyPSA-based network.
+    network : a PyPSA network object (pypsa.Network())
+        The created PyPSA network.
     """
 
-    # create (empty) network
+    # create (empty) PyPSA network
     network = pypsa.Network()
+    network.set_snapshots(snapshots)
 
 
     # add bus "oil" to network
     network.add("Bus",
                 "oil")
 
 
-    # add generator "oil" to bus "oil" with given capital and marginal costs
+    # add generator "oil" to bus "oil" with associated costs
     network.add("Generator",
                 "oil",
                 bus = "oil",
-                capital_cost = oil_capital_cost,
-                marginal_cost = oil_marginal_cost)
+                p_nom_extendable = True,
+                capital_cost = parameters["oil_capital_cost"],
+                marginal_cost = parameters["oil_marginal_cost"])

```

Comment:
I believe that capital cost for generator "oil" it should be set to Zero, because it is an existing infrastructure.
Marginal cost should b e !=0 ( that is correct)